exporters/elasticsearch-exporter/src/main/java/io/camunda/zeebe/exporter/ElasticsearchClient.java

```diff
@@ -128,21 +128,15 @@ public void flush() {
       throw new ElasticsearchExporterException("Failed to flush bulk", e);
     }
 
-    final var failure = checkBulkResponse(bulkResponse);
-    if (failure.isPresent()) {
-      throw failure.get();
+    if (bulkResponse.hasErrors()) {
+      throw collectBulkErrors(bulkResponse);
```

Comment:
maybe we rename the method then to throwBulkErrors

Reply:
But it doesn't throw it? Or do you mean it should throw it? I thought about it but I felt this was more explicit - when reading the flush method, you can see it will throw an exception and stop on error, whereas the other you'd have to look into it.

I have no strong feelings though to be honest, both seem fine and indicate intent - as long as it's not something like `checkError` or something where it's not clear if it throws or not :sweat_smile:

Reply:
Hm I see. Maybe then createExceptionWith or newElasticException

CollectBulkErrors sounds not really like it would create an return an exception object, tbh :sweat_smile: